cgi-bin/DW/Controller/Admin/Approve.pm

```diff
@@ -0,0 +1,180 @@
+            my $pickrand = int( rand($num) );
+            $vars->{user} = $users->{ $ids[$pickrand] };
+            $vars->{ago}  = LJ::diff_ago_text( $vars->{user}->timecreate );
+            return DW::Template->render_template( 'admin/recent_accounts/user.tt',
+                $vars, { no_sitescheme => 1 } );
+        }
+        else {
+            return error_ml("$scope.error.nousers");
+        }
+    }
+
+    if ( $form_args->{uid} ) {
+        my $u       = LJ::load_userid( $form_args->{uid} );
+        my $success = DW::FormErrors->new;
+        my $value;
+
+        # take action based on form_args - either clear or escalate
+        $value = 0 if $form_args->{yes};
+        $value = 2 if $form_args->{no};
+
+        return error_ml('error.invalidform') unless defined $value;
+
+        my $act = $value ? 'rejected' : 'approved';
+        $success->add( '', ".success.$act", { user => $u->ljuser_display } );
+
+        # reinforce the different colors for rejection vs approval
+        $vars->{ $value ? 'errors' : 'success' } = $success;
+
+        # force lookup in case someone else just changed this - we still show
+        # the success message to the user, but don't perform the actions again
+        delete $u->{"not_approved"};
+        $u->preload_props( { use_master => 1 }, "not_approved" );
+
+        if ( $u->prop('not_approved') && $u->prop('not_approved') == 1 ) {
+            $u->set_prop( not_approved => $value );
+            my $msg = sprintf( "new account %s $act by %s", $u->user, $remote->user );
+            LJ::statushistory_add( $u, $remote, 'approvenew', $msg );
+
+            # I thought about automatically doing the suspension here if
+            # $remote had suspend privs, but I think it's better to give the
+            # user a chance to review their actions in case of fat fingering
+        }
+
+        return DW::Template->render_template( 'admin/recent_accounts/review.tt', $vars );
+    }
+
+    if ( $form_args->{suspend} ) {
+        return $r->redirect("$LJ::SITEROOT/admin/recent_accounts")
+            unless $vars->{can_suspend};
+
+        # get a list of flagged users to review for suspension
+        $vars->{users} = $get_users->( 2, 10 );
+
+        return DW::Template->render_template( 'admin/recent_accounts/suspend.tt', $vars );
+    }
+
+    if ( $form_args->{do_suspend} ) {
+        my @ids = split / /, $form_args->{uids};
+        return error_ml('error.invalidform') unless @ids && $vars->{can_suspend};
+
+        my $users   = LJ::load_userids(@ids);
+        my $success = DW::FormErrors->new;
+        my $count   = 0;
+
+        foreach my $uid (@ids) {
+            my $u = $users->{$uid};
+            next if $u->is_suspended;    # already done
+
+            if ( $form_args->{"user_$uid"} ) {
+                $u->set_suspended( $remote, "from /admin/recent_accounts" );
+                $count++;
+            }
+            else {
+                $u->set_prop( not_approved => 0 );
```

Comment:
I forgot, does zero delete the property? I would rather not have the database fill up with zeros.

Reply:
Yes, zero deletes. (Line 1157 of cgi-bin/LJ/User/Permissions.pm)